packages/core/src/workflow/workflow.ts

```diff
@@ -53,169 +40,15 @@ export class Workflow {
     return this.#steps.has(step);
   }
 
-  #acceptsEvent(step: StepFunction<any>, event: WorkflowEvent): boolean {
-    const eventType = event.constructor as EventTypes;
-    const stepInfo = this.#steps.get(step);
-    if (!stepInfo) {
-      throw new Error(`No method found for step: ${step.name}`);
-    }
-    return stepInfo.inputs.includes(eventType);
-  }
-
-  async *streamEvents(): AsyncGenerator<WorkflowEvent, void> {
-    if (this.#contexts.size > 1) {
-      throw new Error(
-        "This workflow has multiple concurrent runs in progress and cannot stream events. " +
-          "To be able to stream events, make sure you call `run()` on this workflow only once.",
-      );
-    }
-
-    const context = this.#contexts.values().next().value;
-    if (!context) {
-      throw new Error("No active context found for streaming events.");
-    }
-
-    yield* context.streamEvents();
-  }
-
-  validate(): void {
-    if (this.#verbose) {
-      console.log("Validating workflow...");
-    }
-
-    // Check if all steps have outputs defined
-    // precondition for the validation to work
-    const allStepsHaveOutputs = Array.from(this.#steps.values()).every(
-      (stepInfo) => stepInfo.outputs !== undefined,
-    );
-    if (!allStepsHaveOutputs) {
-      throw new Error(
-        "Not all steps have outputs defined. Can't validate. Add the 'outputs' parameter to each 'addStep' method call to do validation",
-      );
-    }
-
-    // input events that are consumed by any step of the workflow
-    const consumedEvents: Set<EventTypes> = new Set();
-    // output events that are produced by any step of the workflow
-    const producedEvents: Set<EventTypes> = new Set([StartEvent]);
-
-    for (const [, stepInfo] of this.#steps) {
-      stepInfo.inputs.forEach((eventType) => consumedEvents.add(eventType));
-      stepInfo.outputs?.forEach((eventType) => producedEvents.add(eventType));
-    }
-
-    // Check if all consumed events are produced
-    const unconsumedEvents = Array.from(consumedEvents).filter(
-      (event) => !producedEvents.has(event),
-    );
-    if (unconsumedEvents.length > 0) {
-      const names = unconsumedEvents.map((event) => event.name).join(", ");
-      throw new Error(
-        `The following events are consumed but never produced: ${names}`,
-      );
-    }
+  run(event: StartEvent<Start> | Start): Context<Start> {
```

Comment:
i like that idea that we return a Context calling `run`

Reply:
when I run i also would like to know the type of the stop event, see https://github.com/run-llama/create-llama/blob/ef070c0b4befd62cae3d50295e89ffdbe41ac854/templates/components/multiagent/typescript/nextjs/route.ts#L34

Reply:
added!